repoman/git/depot_operations.py

```diff
@@ -137,57 +120,40 @@ def _locks_cleanup(self, path):
         """
         Inherited method :func:`~DepotOperations._locks_cleanup`
         """
-        repository = pygit2.Repository(path)
-        index_lock_path = os.path.join(repository.path, 'index.lock')
+        index_lock_path = os.path.join(self._git_path(path), 'index.lock')
         if os.path.exists(index_lock_path):
             os.remove(index_lock_path)
 
     def _clear_working_copy(self, path):
-        repository = pygit2.Repository(path)
-        if repository.head_is_unborn:
-            for dirty_file in os.listdir(path):
-                if dirty_file != '.git':
-                    dirty_file_path = os.path.join(path, dirty_file)
-                    if os.path.isdir(dirty_file_path):
-                        shutil.rmtree(dirty_file_path)
-                    else:
-                        os.remove(dirty_file_path)
-        else:
-            repository.reset(repository.head.target, pygit2.GIT_RESET_HARD)
-            repository.checkout_head(strategy=(
-                pygit2.GIT_CHECKOUT_FORCE |
-                pygit2.GIT_CHECKOUT_REMOVE_UNTRACKED))
+        sh.git("reset", "--hard", _cwd=path)
```

Comment:
I've seen some working copies with the content of .git/HEAD `refs/heads/master`. And indeed , that repository had no master branch. Therefore, it caused that `git reset --hard` command failed.

It's high likely that this error was caused due to how main cache and working copies where initialized previously. They just were initialized with init() and then fetched new references. Thereby, they did not have the remote default branch.

Reply:
Ummm, it's perfectly fine for git to have a HEAD with a reference to a non-existing branch, `git reset --hard` works and a commit will create a commit without any parent. There are few cases where the command `git reset --hard` doesn't work, it works even on empty repositories.

But the content of `.git/HEAD` has to be `ref: refs/heads/somebranch` if the head is attached, or a sha1 if the head is detached. If we leave a repository with an invalid content in HEAD we are breaking it and it should be considered a bug in repoman.